src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +56,50 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+    /**
+     * Sets a prompt message as suppressed.
+     * @param promptName Name of prompt to append to list
+     */
+    public disable(promptName: string): void {
```

Comment:
- disable() is too vague of a name, it lives on a generic Settings class. It is suppressing prompts.
- as suggested in https://github.com/aws/aws-toolkit-vscode/pull/1937#discussion_r687092120 , this could be parameterized at near-zero cost, so we should do it to avoid potential refactor in the future.

```suggestion
    public enablePrompt(promptName: string, enable: boolean): void {
```

Reply:
That's what it was called previously when it was in a separate module and you made this comment [here](https://github.com/aws/aws-toolkit-vscode/pull/1937#discussion_r687092120) but now that it's in this class it makes sense to call it something more specific.

Reply:
> you made this comment [here](https://github.com/aws/aws-toolkit-vscode/pull/1937#discussion_r687092120)

by 'generalized to "disable" or "enable"' I meant the function could be generalized to perform both operations. Instead of only performing disable. I didn't mean to change the name.

Reply:
Thanks for the clarification.